src/shared/services/bookmarks-views-plays-service.ts

```diff
@@ -0,0 +1,329 @@
+				isBookmarked
+					? i18n.t('Het aanmaken van de bladwijzer is mislukt')
+					: i18n.t('Het verwijderen van de bladwijzer is mislukt')
+			);
+			return false;
+		}
+	}
+
+	/**
+	 * Gets all bookmarks for user without pagination
+	 * since we cannot order items across both tables: item_bookmarks and collection_bookmarks
+	 * @param user
+	 */
+	public static async getAllBookmarksForUser(user: Avo.User.User): Promise<BookmarkInfo[]> {
+		const response: GetBookmarksForUserResponse = await dataService.query({
+			query: GET_BOOKMARKS_FOR_USER,
+			variables: { profileId: get(user, 'profile.id') },
+		});
+		const itemBookmarks: AppItemBookmark[] = get(response, 'data.app_item_bookmarks', []);
+		const collectionBookmarks: AppCollectionBookmark[] = get(
+			response,
+			'data.app_collection_bookmarks',
+			[]
+		);
+		const itemViews = await this.getMultipleViewCounts(
+			itemBookmarks.map(b => b.item_id),
+			'item'
+		);
+		const collectionViews = await this.getMultipleViewCounts(
+			collectionBookmarks.map(b => b.collection_uuid),
+			'collection'
+		);
+		const itemBookmarkInfos: (BookmarkInfo | null)[] = itemBookmarks.map(
+			(itemBookmark): BookmarkInfo | null => {
+				if (!itemBookmark.bookmarkedItem) {
+					return null;
+				}
+				return {
+					contentId: itemBookmark.item_id,
+					contentType: 'item',
+					createdAt: normalizeTimestamp(itemBookmark.created_at)
+						.toDate()
+						.getTime(),
+					contentTitle: itemBookmark.bookmarkedItem.title,
+					contentThumbnailPath: itemBookmark.bookmarkedItem.thumbnail_path,
+					contentCreatedAt: normalizeTimestamp(itemBookmark.bookmarkedItem.issued)
+						.toDate()
+						.getTime(),
+					contentViews: itemViews[itemBookmark.item_id] || 0,
+				};
+			}
+		);
+		const collectionBookmarkInfos: (BookmarkInfo | null)[] = collectionBookmarks.map(
+			(collectionBookmark): BookmarkInfo | null => {
+				if (!collectionBookmark.bookmarkedCollection) {
+					return null;
+				}
+				return {
+					contentId: collectionBookmark.collection_uuid,
+					contentType:
+						collectionBookmark.bookmarkedCollection.type_id ===
+						ContentTypeNumber.collection
+							? 'collection'
+							: 'bundle',
+					createdAt: normalizeTimestamp(collectionBookmark.created_at)
+						.toDate()
+						.getTime(),
+					contentTitle: collectionBookmark.bookmarkedCollection.title,
+					contentThumbnailPath: collectionBookmark.bookmarkedCollection.thumbnail_path,
+					contentCreatedAt: normalizeTimestamp(
+						collectionBookmark.bookmarkedCollection.created_at
+					)
+						.toDate()
+						.getTime(),
+					contentViews: collectionViews[collectionBookmark.collection_uuid] || 0,
+				};
+			}
+		);
+		return [...compact(itemBookmarkInfos), ...compact(collectionBookmarkInfos)];
+	}
+
+	public static async getMultipleViewCounts(
+		contentIds: string[],
+		type: EventContentTypeSimplified
+	): Promise<{ [uuid: string]: number }> {
+		const response = await dataService.query({
```

Comment:
Idem?

Reply:
die exceptie wordt opgevangen in Bookmarks.tsx, Bookmarks.tsx en BundleDetail.tsx aangezien de error berichten steeds anders zijn